### PR TITLE
Added additional numbers for MIR cards

### PR DESCRIPTION
**Pull request by RavishMan**
## Description

- Added additional checks on the MIR card Aliev Seymur 3 minutes ago

Fixes # (issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### Changed

- Added additional numbers for MIR cards
- Now additional numbers are displayed in a separate fifth block

### CODE REVIEW

File: src/mixins/commonMixin.js [hunk 1]
```diff
@@ -158,8 +159,30 @@ export default {
...
             if (event.isTrusted) return;
             this.$emit(`input-${camelToKebab(type)}`, event.target.value);
+            let validForNextStep = false;
+
+            if (BRANDS_WITH_MULTIPLE_MASKS.includes(this.cardInfo.brandAlias)) {
+                const longest = this.cardInfo.allMasks.sort(function (a, b) {
+                    return b.length - a.length;
+                });
+
+                const MAX_ALLOWED_SYMBOLS = longest[0].replace(
```
Comment (fessracha) [inline src/mixins/commonMixin.js:169]: Use UPPER_CASE_SNAKE_CASE only for static data, here you are getting data dynamically

File: src/consts/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const BRANDS_WITH_MULTIPLE_MASKS = [null, "mir"]; // null - is undefined brand
```
Comment (fessracha) [inline src/consts/index.js:1]: Change null to undefined-bank

File: src/mixins/commonMixin.js [hunk 3]
```diff
@@ -158,8 +159,30 @@ export default {
         onInput(event, type) {
             if (event.isTrusted) return;
             this.$emit(`input-${camelToKebab(type)}`, event.target.value);
+            let validForNextStep = false;
+
+            if (BRANDS_WITH_MULTIPLE_MASKS.includes(this.cardInfo.brandAlias)) {
+                const longest = this.cardInfo.allMasks.sort(function (a, b) {
```
Comment (fessracha) [inline src/mixins/commonMixin.js:165]: move to getMostLongestMask

File: src/services/card-info/brands/schemas/mir.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ export default {
     name: "MIR",
     codeName: "CVC",
     codeLength: 3,
-    gaps: [4, 8, 12],
-    lengths: [16],
-    pattern: /^22\d*$/
+    gaps: [4, 8, 12, 16],
+    lengths: [16, 19],
+    pattern: /^22\d*$/,
```
Comment (fessracha) [inline src/services/card-info/brands/schemas/mir.js:8]: check usage pattern

File: src/services/card-info/index.js [hunk 5]
```diff
@@ -157,6 +188,27 @@ export default class CardInfo extends Card {
         return mask;
     }
 
+    _getAllMasks(maskDigitSymbol, maskDelimiterSymbol, allBlocks) {
```
Comment (fessracha) [inline src/services/card-info/index.js:191]: In my opinion, implementation with reduce will be better.

`const cells = [4, 4, 4, 4]

const mask = cells.reduce((prevCell, cellCharsLength, idx) => {
    const cellMask = prevCell + Array(cellCharsLength + 1).join('#');
    const delimiter = ' ';
    return idx < cells.length - 1  ? `${cellMask}${delimiter}` : cellMask;
}, '')`

File: src/mixins/commonMixin.js (hunk 1 above)
Comment (RavishMan) [reply src/mixins/commonMixin.js:169]: \+

File: src/consts/index.js (hunk 2 above)
Comment (RavishMan) [reply src/consts/index.js:1]: \+

File: src/mixins/commonMixin.js (hunk 3 above)
Comment (RavishMan) [reply src/mixins/commonMixin.js:165]: \+

File: src/services/card-info/brands/schemas/mir.js (hunk 4 above)
Comment (RavishMan) [reply src/services/card-info/brands/schemas/mir.js:8]: is OK

File: src/services/card-info/index.js (hunk 5 above)
Comment (RavishMan) [reply src/services/card-info/index.js:191]: \+

File: src/utils/validators.js [hunk 6]
```diff
@@ -17,10 +18,18 @@ export const length = type =>
...
             return true;
         }
-        return (
-            value.replace(/\s/g, "").length ===
-            vm[type].replace(/\s/g, "").length
-        );
+
+        const valueSymbolLength = value.replace(/\s/g, "").length;
+        // This brands has custom of count numbers on card number
+        if (BRANDS_WITH_MULTIPLE_MASKS.includes(vm.cardInfo.brandAlias) && type === 'cardNumberMask') {
```
[the remark this answers is not in the document]
Comment (RavishMan) [reply src/utils/validators.js:24]: \+

File: tests/services/card-info/card-info.test.js [hunk 7]
```diff
@@ -90,11 +145,191 @@ describe("Card-info", () => {
...
+                    };
+
+                    testValues(colorInfo);
+                });
+            });
+        }
+    );
+
+    describe.each(undefinedCards)(
+        "Check card with number",
```
[the remark this answers is not in the document]
Comment (RavishMan) [reply tests/services/card-info/card-info.test.js:249]: \+

File: src/mixins/commonMixin.js [hunk 8]
```diff
@@ -161,25 +162,21 @@ export default {
...
+            const isMultipleMasks = BRANDS_WITH_MULTIPLE_MASKS.includes(
+                this.cardInfo.brandAlias
+            );
 
-                const inputSymbolCount = event.target.value.replace(
-                    /\s/g,
-                    ""
-                ).length;
+            console.log("isMultipleMasks");
+            console.log(isMultipleMasks);
```
Comment (fessracha) [inline src/mixins/commonMixin.js:175]: remove

**Pull request merged**